### DISCUSSION TITLE
OnFocus modification of multiple TreeSelect components used in react-querybuilder

### DISCUSSION THREAD

**Discussion by JL45312** [primefaces/community#1946] [PrimeReact]
My project is using primereact V8

background: I am currently working on a project that uses reacte-querybuilder with a custom valueEditor. The custom value editor uses TreeSelect with a filter. 

Issue: If I have opened one TreeSelect component, and go to open another, the previous one that I opened does not close. 

What I tried: Using React useRef, and passing it as a prop to all the treeSelect components generated

Is there a way to resolve this issue? Previously I saw that ref was not available in v8 but then I saw a github pr/issue that said it resolved the ref exposure issue

**Comment by jakeboone02** [comment]
Maintainer of `react-querybuilder` here. I'd be happy to take a look at your issue if you put together a sandbox demonstrating it. You can start with [this template](https://react-querybuilder.js.org/sandbox).
